service/Abstractions/Models/DataPipelineStatus.cs

```diff
@@ -67,4 +67,18 @@ public class DataPipelineStatus
     [JsonPropertyOrder(17)]
     [JsonPropertyName("completed_steps")]
     public List<string> CompletedSteps { get; set; } = new();
+
+    /// <summary>
+    /// The failed step, if any.
+    /// </summary>
+    [JsonPropertyOrder(18)]
+    [JsonPropertyName("failed_step")]
+    public string? FailedStep { get; set; } = null;
+
+    /// <summary>
+    /// The error that caused the pipeline to fail, if any.
+    /// </summary>
+    [JsonPropertyOrder(19)]
+    [JsonPropertyName("logs")]
+    public string? Logs { get; set; } = null;
```

Comment:
I would avoid the "string" type which limits to a single unstructured entry, and use `List<PipelineLogEntry> LogEntries` if this is really needed. Thinking about the scenarios, I still don't see a use case though.